debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnectorTask.java

```diff
@@ -202,6 +206,11 @@ public void stop() {
         return replContext.configureLoggingContext(CONTEXT_NAME);
     }
 
+    private void failedReplicator(Throwable t) {
+        replicatorError = t;
+        stop();
```

Comment:
The `stop()` call shouldn't be needed, it'll be triggered by KC once the exception has been re-thrown by `poll()`. While it likely won't do harm, I'd prefer to keep it as simple as possible for the sake of understandability of the overall flow.

Reply:
I tend to disagree here - i went throught the KC code and I believe `stop()` is not called after `poll()` fails but it is vice-versa - `poll()` terminates when the `stop()` is invoked. Could you please double check it?

Reply:
Indeed, good catch!